castle/cms/browser/security/login.py

```diff
@@ -295,9 +296,29 @@ def send_authorization(self):
             'state': new_state
         })
 
+    def check_whitelist(self):
+        username = self.username
+        try:
+            white_listed = api.portal.get_registry_record('whitelist_login_domains')
+            if white_listed:
+                for wl in white_listed:
+                    if wl in username:
```

Comment:
should probably check the end of the username, including the "@" separator for a whilelisted domain -- we specifically want to check for email address like usernames with domains in them here.